ocaml/idl/datamodel_types.ml

```diff
@@ -510,6 +510,9 @@ type obj = {
   obj_doc_tags: doc_tag list;
 } [@@deriving rpc]
 
+let hash_of_obj x =
+  rpc_of_obj x |> Jsonrpc.to_string |> Hashtbl.hash
```

Comment:
Hashtbl.hash can change between OCaml versions, perhaps use `Digest.string` instead

Reply:
Perhaps sorting might be useful.
See  this code from the testsuite:
```
 let rec canonicalise r =
    match r with
>>  | Rpc.Dict x ->
      Rpc.Dict
        (List.sort (fun (s1, _) (s2, _) -> compare s1 s2) x
        |> List.map (fun (x, y) -> x, canonicalise y))
    | Rpc.Enum ys -> Rpc.Enum (List.map canonicalise ys)
    | x -> x
```

Reply:
I'm not convinced this is so useful - why would you want to reorder the fields (other than if you're adding a new field anyway)? 

If you do want to do some stylistic change to the datamodel, then you can do it in one go and then you only have to add a single new hash to the `hash_to_version` map

Reply:
The order of fields is actually important, and shouldn't change.